barretenberg/cpp/src/barretenberg/proof_system/relations/non_native_field_relation.hpp

```diff
@@ -0,0 +1,286 @@
+#pragma once
+#include "barretenberg/ecc/curves/bn254/bn254.hpp"
+#include "barretenberg/numeric/uintx/uintx.hpp"
+#include "relation_parameters.hpp"
+#include "relation_types.hpp"
+
+namespace proof_system {
+
+template <typename FF_> class GoblinTranslatorNonNativeFieldRelationImpl {
+  public:
+    using FF = FF_;
+
+    // 1 + polynomial degree of this relation
+    static constexpr std::array<size_t, 3> SUBRELATION_LENGTHS{
+        3, // Lower wide limb subrelation (checks result is 0 mod 2¹³⁶)
```

Comment:
I don't understand this "wide limb" terminology, could you briefly explain somewhere in there?

Reply:
In a nutshell, its' because we are computing products of limbs and storing that product in there, so it's no longer 68 bits. I added the explanation in the block.